src/reqwest.rs

```diff
@@ -68,30 +65,26 @@ impl Client {
     }
 
     fn next_request(&mut self) -> Result<()> {
-        let mut headers = Headers::new();
-        headers.set(
-            Accept(vec![
-                   qitem(mime::TEXT_EVENT_STREAM),
-            ])
-        );
+        let mut headers = HeaderMap::with_capacity(2);
+        headers.insert(ACCEPT, HeaderValue::from_str("text/event-stream").unwrap());
         if let Some(ref id) = self.last_event_id {
-            headers.set_raw("Last-Event-ID", vec![id.as_bytes().to_vec()]);
+            headers.insert("Last-Event-ID", HeaderValue::from_str(id).unwrap());
         }
 
-        let res = self.client.get(self.url.clone())
-            .headers(headers)
-            .send()?;
+        let res = self.client.get(self.url.clone()).headers(headers).send()?;
 
         // Check status code and Content-Type.
         {
             let status = res.status();
             if !status.is_success() {
                 return Err(ErrorKind::Http(status.clone()).into());
             }
-            if let Some(&ContentType(ref content_type)) = res.headers().get::<ContentType>() {
-                // Compare type and subtype only, MIME parameters are ignored.
-                if (content_type.type_(), content_type.subtype()) != (mime::TEXT, mime::EVENT_STREAM) {
-                    return Err(ErrorKind::InvalidContentType(content_type.clone()).into());
+
+            if let Some(content_type) = res.headers().get(CONTENT_TYPE) {
+                if content_type != "text/event-stream" {
+                    return Err(ErrorKind::InvalidContentType(
+                        content_type.to_str().unwrap().to_string(),
```

Comment:
```suggestion
                        String::from_utf8_lossy(content_type.as_bytes()).into_owned(),
```

would help you avoid a panic.